File: infopedia-subcontent/src/main/java/com/iorix2k2/infopedia/subcontent/model/SubContentType.java
package com.iorix2k2.infopedia.subcontent.model;

import org.apache.commons.lang.StringUtils;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;


public enum SubContentType
{
	TEXT("text"),
	IMAGE_LINK("image-link"),
	LINK("link");
	
	
	private SubContentType(String text)
	{
		this.text = text;
	}
	
	@JsonValue
	public String getText()
	{
		return text;
	}

	@JsonCreator
	public static SubContentType createFromText(String text)
	{
		var t = StringUtils.defaultString(text).toLowerCase().trim();
		
		if(t.equals("text"))
			return SubContentType.TEXT;
		else if(t.equals("image-link"))
			return SubContentType.IMAGE_LINK;
		else if(t.equals("link"))
			return SubContentType.LINK;
		
		return null;
	}
	
	
	private String text;
}

File: infopedia-frontend/src/main/webapp/WEB-INF/frontend/src/model/SubContentType.ts
export enum SubContentType {
  text = "text",
  imageLink = "image-link",
  link = "link"
}

class SubContentTypeUtil {
  public equals = (sct1: SubContentType, sct2?: SubContentType) => {
    return sct1 === sct2;
  }

  public createFrom(text?: string) {
    const t = text?.toLowerCase();
    return text === "image-link" ? SubContentType.imageLink :
        text === "link" ? SubContentType.link : SubContentType.text;
  }

  public isValid(text?: string) {
    const t = text?.toLowerCase();
    return t === "text" || t === "image-link" || t === "link";
  }

  public getValues() {
    return [SubContentType.text, SubContentType.imageLink, SubContentType.link];
  }
}


export const subContentTypeUtil = new SubContentTypeUtil();
export default SubContentType;